### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading 'reduce')

### DISCUSSION THREAD

**Discussion by KivancUlucan** [neuefische/questions#609] [Web]
I would like to have the number of created tasks (broken down by category) displayed on pages/index.js at ` <SummaryNumber>{count}</SummaryNumber>` instead of the hardcoded number. I have now tried for a while to solve this, but unfortunately do not come to the solution. 

I have committed and pushed my current [status](https://github.com/KivancUlucan/Capstone-Project/pull/11)  The branch is called counts-of-tasks.
Thanks

**Comment by ThomasF85** [comment]
Hi Kivanc,

your SummaryCard component expects a prop called `tasks`:

```
export default function SummaryCard({ text, size, tasks }) {
  const [categoryCounts, setCategoryCounts] = useState({});

  useEffect(() => {
    const counts = tasks.reduce((counts, task) => {
      counts[task.category] = (counts[task.category] || 0) + 1;
      return counts;
    }, {});

    setCategoryCounts(counts);
  }, [tasks]);
```

This prop has to be handed over to the component or else it will be undefined. You are not handing the prop over, see for example your `pages/index.js`:

`<SummaryCard text="Tasks in Backlog" size="normal" />`

**Comment by KivancUlucan** [comment]
I passed the prop as following:

 <SummaryCard text="Tasks in Backlog" size="normal" tasks={tasks} />
 
 now i have this error allert
 
 `Server Error
ReferenceError: tasks is not defined

This error happened while generating the page. Any console logs will be displayed in the terminal window.
Source
pages/index.js (9:66) @ tasks

   7 | <h1>Summary</h1>
   8 | <Section1>
>  9 |   <SummaryCard text="Tasks in Backlog" size="normal" tasks={tasks} />
     |                                                            ^
  10 |   <SummaryCard text="Tasks in Progress" size="normal" tasks={tasks} />
  11 | </Section1>
  12 | <Section2>`
  
 
This is the [PR](https://github.com/KivancUlucan/Capstone-Project/pull/11/files)

**Reply by ThomasF85** [reply]
You are passing the variable tasks in your index.js, but the variable is undefined. You can only use variable names that exist. You have to somehow get your tasks from somewhere (state, localStorage, etc. whatever your data source is).
